Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean

```diff
@@ -483,6 +483,37 @@ theorem inv_eq_left_inv (h : B * A = 1) : A⁻¹ = B :=
 theorem inv_eq_right_inv (h : A * B = 1) : A⁻¹ = B :=
   inv_eq_left_inv (mul_eq_one_comm.2 h)
 
+section Woodbury
+
+variable [Fintype m] [DecidableEq m]
+variable (A: Matrix n n α) (U : Matrix n m α) (C : Matrix m m α) (V : Matrix m n α)
+variable [Invertible A] [Invertible C] [Invertible (C⁻¹ + V * A⁻¹* U)]
```

Comment:
Can you restate this to use `⅟` instead of `⁻¹`?
```suggestion
variable [Invertible A] [Invertible C] [Invertible (⅟C + V * ⅟A * U)]
```
I think this would alllow it to generalize to matrices over non-commutative rings; and this would fix @vihdzp's `noncomputable` comment below.

Reply:
The proof goes through as
```
ble [Invertible A] [Invertible C] [Invertible (⅟C + V * ⅟A* U)]

lemma add_mul_mul_inv_mul_eq_one: (A+U*C*V)*(⅟A-⅟A*U*⅟(⅟C+V*⅟A*U)*V*⅟A) = 1 := by
  calc
      (A+U*C*V)*(⅟A-⅟A*U*⅟(⅟C+V*⅟A*U)*V*⅟A)
      _ = A*⅟A-A*⅟A*U*⅟(⅟C+V*⅟A*U)*V*⅟A+U*C*V*⅟A-U*C*V*⅟A*U*⅟(⅟C+V*⅟A*U)*V*⅟A := by
        simp_rw [add_sub_assoc, Matrix.add_mul, Matrix.mul_sub, Matrix.mul_assoc]
      _ = (1+U*C*V*⅟A)-(U*⅟(⅟C+V*⅟A*U)*V*⅟A+U*C*V*⅟A*U*⅟(⅟C+V*⅟A*U)*V*⅟A) := by
        rw [mul_invOf_self, Matrix.one_mul]
        abel
      _ = 1+U*C*V*⅟A-(U+U*C*V*⅟A*U)*⅟(⅟C+V*⅟A*U)*V*⅟A := by
        rw [sub_right_inj, Matrix.add_mul, Matrix.add_mul, Matrix.add_mul]
      _ = 1+U*C*V*⅟A-U*C*(⅟C+V*⅟A*U)*⅟(⅟C+V*⅟A*U)*V*⅟A := by
        congr
        simp only [Matrix.mul_add, Matrix.mul_mul_invOf_self_cancel, ← Matrix.mul_assoc]
      _ = 1 := by
        rw [Matrix.mul_mul_invOf_self_cancel]
        abel
```
which is essentially the original proof with some lemma renames.